presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/rtree/TestFlatbush.java

```diff
@@ -0,0 +1,300 @@
+    }
+
+    @Test
+    public void testOctagonQuery()
+    {
+        OGCGeometryWrapper octagonA = new OGCGeometryWrapper(POLYGON_A);
+        OGCGeometryWrapper octagonB = new OGCGeometryWrapper(POLYGON_B);
+
+        OGCGeometryWrapper pointX = new OGCGeometryWrapper(POINT_X);
+        OGCGeometryWrapper pointY = new OGCGeometryWrapper(POINT_Y);
+        OGCGeometryWrapper pointZ = new OGCGeometryWrapper(POINT_Z);
+        OGCGeometryWrapper pointW = new OGCGeometryWrapper(POINT_W);
+
+        Flatbush<OGCGeometryWrapper> rtree = new Flatbush<>(new OGCGeometryWrapper[] {pointX, pointY, pointZ, pointW});
+
+        List<OGCGeometryWrapper> resultsA = findIntersections(rtree, octagonA.getExtent());
+        assertEqualsSorted(resultsA, Arrays.asList(pointX, pointY), Comparator.naturalOrder());
+
+        List<OGCGeometryWrapper> resultsB = findIntersections(rtree, octagonB.getExtent());
+        assertEqualsSorted(resultsB, Arrays.asList(pointY, pointZ), Comparator.naturalOrder());
+    }
+
+    @Test
+    public void testOctagonTree()
+    {
+        OGCGeometryWrapper octagonA = new OGCGeometryWrapper(POLYGON_A);
+        OGCGeometryWrapper octagonB = new OGCGeometryWrapper(POLYGON_B);
+
+        OGCGeometryWrapper pointX = new OGCGeometryWrapper(POINT_X);
+        OGCGeometryWrapper pointY = new OGCGeometryWrapper(POINT_Y);
+        OGCGeometryWrapper pointZ = new OGCGeometryWrapper(POINT_Z);
+        OGCGeometryWrapper pointW = new OGCGeometryWrapper(POINT_W);
+
+        Flatbush rtree = new Flatbush<OGCGeometryWrapper>(new OGCGeometryWrapper[] {octagonA, octagonB});
+
+        assertEquals(findIntersections(rtree, pointX.getExtent()), ImmutableList.of(octagonA));
+
+        List<OGCGeometryWrapper> results = findIntersections(rtree, pointY.getExtent());
+        results.sort(Comparator.naturalOrder());
+        assertEquals(results, ImmutableList.of(octagonA, octagonB));
+
+        assertEquals(findIntersections(rtree, pointZ.getExtent()), ImmutableList.of(octagonB));
+
+        assertEquals(findIntersections(rtree, pointW.getExtent()), ImmutableList.of());
+    }
+
+    @DataProvider(name = "rectangle-counts")
+    private Object[][] rectangleCounts()
+    {
+        return new Object[][] {{100, 1000, 42}, {1000, 10_000, 123}, {5000, 50_000, 321}};
+    }
+
+    @Test(dataProvider = "rectangle-counts")
+    public void testRectangleCollection(int numBuildRects, int numProbeRects, int seed)
+    {
+        Random random = new Random(seed);
+        List<Rectangle> buildRects = makeRectangles(random, numBuildRects);
+        List<Rectangle> probeRects = makeRectangles(random, numProbeRects);
+
+        Flatbush rtree = new Flatbush<Rectangle>(buildRects.toArray(new Rectangle[] {}));
+        for (Rectangle query : probeRects) {
+            List<Rectangle> rtreeResults = findIntersections(rtree, query);
+            List<Rectangle> bruteResults = buildRects.stream()
```

Comment:
use canonical names: actual and expected

(or spell out brute force: bruteForceResults)